WooCommerce/Classes/Tools/BackgroundTasks/ForegroundPOSCatalogSyncDispatcher.swift

```diff
@@ -0,0 +1,210 @@
+        static let syncInterval: TimeInterval = 60 * 60 // 1 hour
+        static let initialSyncDelay: TimeInterval = 5 // 5 seconds after becoming active
+        static let leeway: Int = 5 // 5 seconds leeway to give a system more flexibility in managing resources
+    }
+
+    private let interval: TimeInterval
+    private let notificationCenter: NotificationCenter
+    private let timerProvider: DispatchTimerProviding
+    private let featureFlagService: FeatureFlagService
+    private let stores: StoresManager
+    private var observers: [NSObjectProtocol] = []
+    private var timer: DispatchTimerProtocol?
+
+    private var isRunning: Bool {
+        syncSiteID != nil
+    }
+    private var syncSiteID: Int64?
+
+    init(interval: TimeInterval = Constants.syncInterval,
+         notificationCenter: NotificationCenter = .default,
+         timerProvider: DispatchTimerProviding = DefaultDispatchTimerProvider(),
+         featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService,
+         stores: StoresManager = ServiceLocator.stores) {
+        self.interval = interval
+        self.notificationCenter = notificationCenter
+        self.timerProvider = timerProvider
+        self.featureFlagService = featureFlagService
+        self.stores = stores
+    }
+
+    deinit {
+        stop()
+    }
+
+    func start() {
+        guard featureFlagService.isFeatureFlagEnabled(.pointOfSaleLocalCatalogi1) else {
+            return
+        }
+
+        if syncSiteID != stores.sessionManager.defaultStoreID {
+            DDLogInfo("🔄 ForegroundPOSCatalogSyncDispatcher: Site has changed, resetting the sync")
+            stop()
+        }
+
+        guard !isRunning else { return }
+
+        DDLogInfo("🔄 ForegroundPOSCatalogSyncDispatcher: Starting foreground sync dispatcher")
+
+        let activeObserver = notificationCenter.addObserver(
+            forName: UIApplication.didBecomeActiveNotification,
+            object: nil,
+            queue: .main
+        ) { [weak self] _ in
+            self?.startTimer()
+        }
+
+        let backgroundObserver = notificationCenter.addObserver(
+            forName: UIApplication.didEnterBackgroundNotification,
+            object: nil,
+            queue: .main
+        ) { [weak self] _ in
+            self?.stopTimer()
+        }
+
+        let defaultSiteObserver = notificationCenter.addObserver(
+            forName: .StoresManagerDidUpdateDefaultSite,
+            object: nil,
+            queue: .main
+        ) { [weak self] _ in
+            self?.stop()
+        }
+
+        observers = [activeObserver, backgroundObserver, defaultSiteObserver]
+
+        if UIApplication.shared.applicationState == .active {
+            startTimer()
+        }
+
+        syncSiteID = stores.sessionManager.defaultStoreID
+    }
+
+    func stop() {
+        guard isRunning else { return }
+
+        DDLogInfo("🛑 ForegroundPOSCatalogSyncDispatcher: Stopping foreground sync dispatcher")
```

Comment:
Consider if passing the `siteID` in the DDLog here would be useful: When I switched from a POS-eligible store to a non-eligible one the only notice logged was that the timer stopped. Perhaps not necessary as we would see the siteID changing from context of other events.